src/odemis/acq/align/shift.py

```diff
@@ -19,167 +19,158 @@
     """
-    Given two images, it calculates the shift in x and y axis. It first computes
-    the cross-correlation of the two images and then locates the peak. The coordinates
-    of the peak of the cross-correlation define the shift vector between the two images.
-    This implementation is based on the "Efficient subpixel image registration by
+    The function is taken from skimage.registration._phase_cross_correlation.
+    Given two images, it calculates the shift in x and y axis. It first computes the cross-correlation of the
+    two images. Then it modifies cross-correlation values
+    by reducing the amplitude of noise and enhancing the shift amplitude. The coordinates
+    of the maximum of the cross-correlation define the shift vector between the two images.
+    The implementation of precision > 1 is based on the "Efficient subpixel image registration by
     cross-correlation" by Manuel Guizar, for the corresponding matlab code see
     http://www.mathworks.com/matlabcentral/fileexchange/
     18401-efficient-subpixel-image-registration-by-cross-correlation.
 
+    For e.g.
+    When the current image moves to top right corner relatively to the previous image.
+    The returned shift value is positive in both x and y respectively.
+
     previous_img (numpy.array): 2d array with the previous frame
     current_img (numpy.array): 2d array with the last frame, must be of same
       shape as previous_img
     precision (1<=int): Calculate drift within 1/precision of a pixel
-    returns (tuple of floats): Drift in pixels
+    returns (tuple of floats): Drift in pixels (horizontal, vertical).
     """
     if precision < 1:
         raise ValueError("Precision cannot be less than 1, got %s." % (precision,))
-    assert previous_img.shape == current_img.shape, "Prev shape %s != new shape %s" % (previous_img.shape, current_img.shape)
+    assert previous_img.shape == current_img.shape, "Prev shape %s != new shape %s" % (
+        previous_img.shape, current_img.shape)
 
     previous_fft = fft.fft2(previous_img)
     current_fft = fft.fft2(current_img)
     m, n = previous_fft.shape
-
-    if precision == 1:
-        # Cross-correlation computation
-        CC = fft.ifft2(previous_fft * current_fft.conj())
-
-        # Locate the peak
-        ACC = abs(CC)
-        loc1 = ACC.argmax(0)
-        max1 = ACC[(loc1, range(ACC.shape[1]))]
-        loc2 = max1.argmax(0)
-
-        rloc = loc1[loc2]
-        cloc = loc2
-
-        # Calculate shift from the peak
-        md2 = m // 2
-        nd2 = n // 2
-        if rloc > md2:
-            row_shift = rloc - m
-        else:
-            row_shift = rloc
-
-        if cloc > nd2:
-            col_shift = cloc - n
-        else:
-            col_shift = cloc
-
+    image_product = previous_fft * current_fft.conj()
+
+    # Cross-correlation computation
+    eps = numpy.finfo(image_product.real.dtype).eps
+    image_product /= numpy.maximum(numpy.abs(image_product), 100 * eps)
```

Comment:
Please explain the magical part of this block, or at least put a reference to where this can be explained.